### DISCUSSION TITLE
Is this feature from Open AI supported by Semantic Kernel - force the model to always call one or more functions, you can set tool_choice: "required".

### DISCUSSION THREAD

**Discussion by EVENFLOW212** [microsoft/semantic-kernel#6971] [Q&A] [answered]
https://platform.openai.com/docs/guides/function-calling/function-calling-behavior

Is the below feature supported by Semantic Kernel?

To force the model to always call one or more functions, you can set tool_choice: "required". The model will then select which function(s) to call.

Thanks

**Comment by dmytrostruk** [comment, marked as answer]
@EVENFLOW212 Yes, it is supported by using `ToolCallBehavior.RequireFunction()` in execution settings:
https://github.com/microsoft/semantic-kernel/blob/d2bc3a83e89f6a1d1a9855a44b9371c80c934082/dotnet/samples/Concepts/Filtering/AutoFunctionInvocationFiltering.cs#L30-L33

As for now it accepts only one function, but we are working on new function calling abstractions that will accept multiple required functions to call. 

Thank you!

**Reply by EVENFLOW212** [reply]
Hi @dmytrostruk 

We have multiple plugins and multiple functions for our client facing app. 

Would you know what the timeline would be for the implementation of multiple required function calls?

Thanks

**Reply by SergeyMenshykh** [reply]
@EVENFLOW212, as @dmytrostruk said, we are currently working on the new function calling model - https://github.com/microsoft/semantic-kernel/pull/6083 - and plan to go live with it around the second half of July.

**Reply by EVENFLOW212** [reply]
Hi @dmytrostruk 

Please let us know when this is implemented. We are using gpt-4o and does not seem that the azure open ai we are using supports this functionality yet.

Thanks

**Reply by dmytrostruk** [reply]
@EVENFLOW212 Multiple required function calls are now supported in latest version of Semantic Kernel. Here is an example, where the configuration accepts an array of functions:
https://github.com/microsoft/semantic-kernel/blob/f08cf3c7780ad02035698c989bb203c6fc54fdb0/dotnet/samples/Concepts/Filtering/AutoFunctionInvocationFiltering.cs#L30-L33
